extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -413,3 +422,116 @@ export async function attemptRerun(
     void showAndLogErrorMessage(error);
   }
 }
+
+/**
+ * Lists the workflow run artifacts for the given workflow run ID.
+ * @param credentials Credentials for authenticating to the GitHub API.
+ * @param owner
+ * @param repo
+ * @param workflowRunId The ID of the workflow run to list artifacts for.
+ * @returns An array of artifact details (including artifact name and ID).
+ */
+async function listWorkflowRunArtifacts(
+  credentials: Credentials,
+  owner: string,
+  repo: string,
+  workflowRunId: number
+) {
+  const octokit = await credentials.getOctokit();
+  const response = await octokit.rest.actions.listWorkflowRunArtifacts({
+    owner,
+    repo,
+    run_id: workflowRunId,
+  });
+
+  return response.data.artifacts;
+}
+
+/**
+ * @param artifactName The artifact name, as a string.
+ * @param artifacts An array of artifact details (from the "list workflow run artifacts" API response).
+ * @returns The artifact ID corresponding to the given artifact name.
+ */
+function getArtifactIDfromName(artifactName: string, artifacts: Array<{ id: number, name: string }>): number | undefined {
+  const artifact = artifacts.find(a => a.name.localeCompare(artifactName) === 0);
```

Comment:
Do we want a case insensitive compare here? I'm assuming `artifactName` is from user input. And if so, I think case-insentive would make a better user experience.

Reply:
That was actually my suggestion 🙈  Sorry @shati-patel - I was under the impression the artifact name is set by the Actions engine.